.github: Build test binaries in build/ directory

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56942 .github: Enable Linux CPU GHA on PRs
* #56945 .github: Bump linux.2xlarge runners to 500
* **#56941 .github: Build test binaries in build/ directory**

Sets the custom test binaries we build in .jenkins/pytorch/build.sh to
be built in the `build` directory instead of the directory above the
workspace.

This should alleviate any weirdness we were seeing before with test
binaries having to be overwritten

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D28018453](https://our.internmc.facebook.com/intern/diff/D28018453)

seemethere committed:
.github: Build test binaries in build/ directory

Sets the custom test binaries we build in .jenkins/pytorch/build.sh to
be built in the `build` directory instead of the directory above the
workspace.

This should alleviate any weirdness we were seeing before with test
binaries having to be overwritten

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Build test binaries in build/ directory"


Sets the custom test binaries we build in .jenkins/pytorch/build.sh to
be built in the `build` directory instead of the directory above the
workspace.

This should alleviate any weirdness we were seeing before with test
binaries having to be overwritten

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Build test binaries in build/ directory"


Sets the custom test binaries we build in .jenkins/pytorch/build.sh to
be built in the `build` directory instead of the directory above the
workspace.

This should alleviate any weirdness we were seeing before with test
binaries having to be overwritten

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Build test binaries in build/ directory"


Sets the custom test binaries we build in .jenkins/pytorch/build.sh to
be built in the `build` directory instead of the directory above the
workspace.

This should alleviate any weirdness we were seeing before with test
binaries having to be overwritten

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

walterddr commented on .jenkins/pytorch/test.sh:
```diff
@@ -7,6 +7,9 @@
 COMPACT_JOB_NAME="${BUILD_ENVIRONMENT}"
 
+# Get fully qualified path using realpath
+CUSTOM_TEST_ARTIFACT_BUILD_DIR=$(realpath ${CUSTOM_TEST_ARTIFACT_BUILD_DIR:-${PWD}/../})
```
nit: why do we need this default back to `${PWD}/../` behavior? can we remove it entirely?

seemethere replied:
I wanted to keep them so that the circleci builds don't actually change their current behavior.

Trying to make the least obstructive change here.

walterddr commented on .jenkins/pytorch/build.sh:
```diff
@@ -240,8 +240,11 @@ else
       cp build/.ninja_log dist
     fi
 
+    CUSTOM_TEST_ARTIFACT_BUILD_DIR=${CUSTOM_TEST_ARTIFACT_BUILD_DIR:-${PWD}/../}
```
ditto

seemethere committed:
Update on ".github: Build test binaries in build/ directory"


Sets the custom test binaries we build in .jenkins/pytorch/build.sh to
be built in the `build` directory instead of the directory above the
workspace.

This should alleviate any weirdness we were seeing before with test
binaries having to be overwritten

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]
